### PR TITLE
[Transformations] ConvertDivide disabled transformation for divide ops from precision sensitive subgraphs

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - *Added markup transformation to mark fp16 divides in shapeOf subgraphs*
 - *Added "nonconvertible_divide" attribute*
 - *ConvertDivide disabled for non-convertible divides*

### Tickets:
 - *57322*

### CODE REVIEW

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 1]
```diff
@@ -176,6 +177,7 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
 
     static const auto precisions = get_convert_precisions();
 
+    manager.register_pass<ov::pass::MarkPrecisionSensitiveDivides>();
```
Comment (jane-intel) [inline src/plugins/intel_cpu/src/mkldnn_plugin.cpp:180]: Lets maybe have it in Common? GPU will need this code also we can add this into Common pipeline enabled by default after the first `ConstantFolding`. Plugins which don't support dynamism will have folded ShapeOf sub-graphs starting after such `ConstantFolding` but CPU (and GPU in the future) would have valid behavior (no Divide decomposition).

File: src/common/transformations/src/transformations/op_conversions/convert_divide.cpp [hunk 2]
```diff
@@ -14,14 +14,17 @@
...
 NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertDivideWithConstant, "ConvertDivideWithConstant", 0);
 
 namespace {
 bool convert_divide(std::shared_ptr<ngraph::Node> node) {
     auto div = std::dynamic_pointer_cast<ngraph::opset1::Divide>(node);
     // We can not apply this transformation in case with integer input data type
-    if (!div || div->get_input_element_type(0).is_integral()
+    if (!div || ov::divide_is_nonconvertible(div)
+             || div->get_input_element_type(0).is_integral()
              || div->get_input_element_type(1).is_integral()) {
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply src/common/transformations/src/transformations/op_conversions/convert_divide.cpp]: Corrected, thanks

File: src/common/transformations/src/transformations/common_optimizations/mark_precision_sensitive_subgraphs.cpp [hunk 3]
```diff
@@ -58,8 +31,15 @@ bool ov::pass::MarkPrecisionSensitiveSubgraphs::run_on_model(const std::shared_p
...
             continue;
         for (auto& input : curr_node->inputs()) {
-            if (ov::is_precision_sensitive(input))
-                visit_shape_path(input.get_source_output().get_node_shared_ptr(), visited);
+            if (ov::is_precision_sensitive(input)) {
+                auto markup_func = [](shared_ptr<Node> node) {
+                    if (ov::is_type<ov::opset8::Constant>(node)) {
+                        ov::disable_fp16_compression(node);
+                    }
+                };
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply src/common/transformations/src/transformations/common_optimizations/mark_precision_sensitive_subgraphs.cpp]: Done, thank you!

File: src/common/transformations/src/transformations/common_optimizations/mark_precision_sensitive_divides.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        nodes.pop_front();
+        if (visited.count(curr_node))
+            continue;
+        for (auto& input : curr_node->inputs()) {
+            if (ov::is_precision_sensitive(input)) {
+                auto markup_func = [](std::shared_ptr<Node> node) {
+                    if (ov::is_type<ov::opset8::Divide>(node) && node->get_output_element_type(0) == ngraph::element::f16) {
+                        ov::disable_divide_conversion(node);
+                    }
+                };
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply src/common/transformations/src/transformations/common_optimizations/mark_precision_sensitive_divides.cpp]: Corrected

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp (hunk 1 above)
Comment (v-Golubev) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: Markup pass has been moved into common optimization pass. Thanks!

**Review by jane-intel — APPROVED** [review]

**Pull request merged**